slash/resuming.py

```diff
@@ -125,21 +125,29 @@ def get_last_resumeable_session_id():
 
 def get_tests_from_previous_session(session_id, get_successful_tests=False):
     unstarted_first, failed_first = config.root.resume.unstarted_first, config.root.resume.failed_first
+    unstarted_only, failed_only = config.root.resume.unstarted_only, config.root.resume.failed_only
     if unstarted_first and failed_first:
         raise CannotResume("Got both failed_first and planned_first, cannot choose")
+    if unstarted_only and failed_only:
+        raise CannotResume("Got both unstarted_only and failed_only, cannot resume")
     with connecting_to_db() as conn:
         # pylint: disable=no-member
         if conn.query(SessionMetadata).filter(SessionMetadata.session_id == session_id).first():
-            session_tests = conn.query(ResumeState).filter(ResumeState.session_id == session_id).all()
+            session_tests_query = conn.query(ResumeState).filter(ResumeState.session_id == session_id)
```

Comment:
I think this code should be reworked at this point. Have a separate helper strategy class/function that yields all tests and their attributes as munches (started=True/False, success=True/False), and then a separate logic that prepares a list to run from them